spec/models/recipe_spec.rb

```diff
@@ -12,4 +12,15 @@
     it {should have_many :recipe_ingredients}
     it {should have_many(:ingredients).through(:recipe_ingredients)}
   end
+
+  describe "total_cost" do
+    it "can calculate the total cost of a recipe" do
+      @recipe = Recipe.create(name: "spaghetti", complexity: 1, genre: "Italian")
+      @ingredient_1 = @recipe.ingredients.create(name: "pasta", cost: 2)
+      @ingredient_2 = @recipe.ingredients.create(name: "sauce", cost: 3)
+      @ingredient_3 = @recipe.ingredients.create(name: "ground beef", cost: 5)
+
+      expect(@recipe.total_cost).to eq(10)
+    end
```

Comment:
This is great. To make this test a little more robust, I would add another recipe and associate some ingredients to it and run `total_cost` on that to see a different number than `10`. This will show the flexibility of your logic code.